codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/NamingObstacleCourseTestModels.kt

```diff
@@ -169,4 +170,56 @@ object NamingObstacleCourseTestModels {
             """,
         )
     }.toString().asSmithyModel()
+
+    fun reusedInputOutputShapesModel(protocol: Trait) = """
+        namespace test
+        use ${protocol.toShapeId()}
+        use aws.api#service
+        @${protocol.toShapeId().name}
+        @service(sdkId: "test")
+        service Service {
+            version: "2006-03-01",
+            operations: [GetThing, GetThingNested]
+        }
+
+        // re-use get thing output in a list & in an operation
+        @http(uri: "/SomeOperation2", method: "POST")
+        operation GetThing {
+            output: GetThingOutput
+            input: GetThingOutput
```

Comment:
```suggestion
            input: GetThingInput
```

Reply:
it actually was on purpose but I've cleaned up the names and comments to clarify whats going on